hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -81,23 +81,29 @@ class HoodieMergeOnReadRDD(@transient sc: SparkContext,
 
   override def compute(split: Partition, context: TaskContext): Iterator[InternalRow] = {
     val partition = split.asInstanceOf[HoodieMergeOnReadPartition]
+    lazy val partitionPath = LogFileIterator.getPartitionPath(partition.split.dataFile, partition.split.logFiles)
     val iter = partition.split match {
       case dataFileOnlySplit if dataFileOnlySplit.logFiles.isEmpty =>
         val projectedReader = projectReader(fileReaders.requiredSchemaReaderSkipMerging, requiredSchema.structTypeSchema)
         projectedReader(dataFileOnlySplit.dataFile.get)
 
       case logFileOnlySplit if logFileOnlySplit.dataFile.isEmpty =>
-        new LogFileIterator(logFileOnlySplit, tableSchema, requiredSchema, tableState, getHadoopConf)
+        new LogFileIterator(logFileOnlySplit.logFiles, partitionPath, tableSchema, requiredSchema,
+          tableState, getHadoopConf)
 
       case split =>
         mergeType match {
           case DataSourceReadOptions.REALTIME_SKIP_MERGE_OPT_VAL =>
             val reader = fileReaders.requiredSchemaReaderSkipMerging
-            new SkipMergeIterator(split, reader, tableSchema, requiredSchema, tableState, getHadoopConf)
+            val iterator = reader(split.dataFile.get)
+            new SkipMergeIterator(split.logFiles, partitionPath, iterator, reader.schema, tableSchema,
```

Comment:
Since Split abstracts the set of files to process, Its better to pass the split directly to different iterators